plugins/pulp_rpm/plugins/importers/iso/importer.py

```diff
@@ -107,27 +108,31 @@ def sync_repo(self, repo, sync_conduit, config):
         self.iso_sync = None
         return report
 
-    def upload_unit(self, repo, type_id, unit_key, metadata, file_path, conduit, config):
+    def upload_unit(self, transfer_repo, type_id, unit_key, metadata, file_path, conduit, config):
         """
         See super(self.__class__, self).upload_unit() for the docblock explaining this method. In
         short, it handles ISO uploads.
         """
-        iso = models.ISO(unit_key['name'], unit_key['size'], unit_key['checksum'])
-        iso.init_unit(conduit)
+        repo = platform_models.Repository.objects.get(repo_id=transfer_repo.id)
```

Comment:
Usually the ``transfer_repo`` has the new ``Repository`` model available on its ``.repo_obj`` attribute. It's worth checking there before loading from the DB.